File: JavaScript/67-Add Binary.js
/**
 * @param {string} a
 * @param {string} b
 * @return {string}
 */
var addBinary = function(a, b) {
    let indexA = a.length-1, indexB = b.length-1;
    let carry = 0;
    let res = "";
    
    while (indexA >= 0 || indexB >= 0){
        if (indexA >= 0){
            if (a[indexA] === '1'){
                carry += 1;
            }
            indexA--;
        }
        
        if (indexB >= 0){
            if (b[indexB] === '1'){
                carry += 1;
            }
            indexB--;
        }       
        
        res = (carry%2).toString() + res;
        carry = ~~(carry/2);
    }
    
    if (carry) res = "1" + res;
    
    return res;
    
};

File: C++/67-Add Binary.cpp
class Solution {
public:
    string addBinary(string a, string b) {

        int idx1 = a.size()-1, idx2 = b.size()-1, carry = 0;
        string res = "";
        
        while (idx1 >= 0 || idx2 >= 0 || carry) {
            if (idx1 >= 0) {
                carry += a[idx1--]-'0';
            }
            if (idx2 >= 0) {
                carry += b[idx2--]-'0';
            }

            res += (carry%2+'0');
            carry = carry/2;
        }

        reverse(begin(res), end(res));
        return res;
    }
};